tools/vargo/src/main.rs

```diff
@@ -679,6 +683,111 @@ fn run() -> Result<(), String> {
+                                    .map(|x| x.as_str())
+                                    .collect::<std::collections::HashSet<&str>>();
+                                if !original_args.is_subset(&std::collections::HashSet::from([
+                                    "--",
+                                    "--check",
+                                    "fmt",
+                                    "--color=always",
+                                ])) {
+                                    return Err(format!(
+                                        "some of the arguments to fmt are unsupported"
+                                    ));
+                                }
+                                original_args.contains("--check")
+                            }
+                            Err(err) => match err.kind() {
+                                std::io::ErrorKind::NotFound => {
+                                    warn(format!("cannot execute verusfmt, refer to CONTRIBUTING.md for installation instructions\nvstd will not be formatted").as_str());
+                                    return Ok(());
+                                }
+                                _ => return Err(format!("cannot execute verusfmt: {}", err)),
+                            },
+                        };
+
+                        info(format!("formatting vstd").as_str());
+
+                        let vstd_path = std::path::Path::new("vstd");
+                        let walk_vstd = walkdir::WalkDir::new(vstd_path)
+                            .follow_links(true)
+                            .into_iter()
+                            .filter_map(|e| e.ok())
+                            .filter(|x| x.path().extension() == Some(std::ffi::OsStr::new("rs")))
+                            .map(|x| x.path().to_owned())
+                            .collect::<Vec<_>>();
+
+                        for vstd_file_path in walk_vstd.iter() {
+                            let mut verusfmt = std::process::Command::new("verusfmt");
+                            if verusfmt_check {
+                                verusfmt.arg("--check");
+                            }
+                            verusfmt.arg(vstd_file_path);
+                            let verusfmt_status = verusfmt.status().expect("failed to run vstd");
+
+                            if !verusfmt_status.success() {
+                                warn(
+                                    format!(
+                                        "verusfmt returned error code {:?} on file {}, exiting",
+                                        verusfmt_status.code(),
+                                        vstd_file_path.display()
+                                    )
+                                    .as_str(),
+                                );
+                                std::process::exit(verusfmt_status.code().unwrap_or(1))
+                            }
+                        }
```

Comment:
`verusfmt` explicitly continues checking files even upon failure (see https://github.com/verus-lang/verusfmt/pull/37), since that leads to better user experience (when there is more than one file that has an issue, it is best to show them all at once, rather than quitting on the first). The easiest way to achieve this would be to simply pass all the arguments to verusfmt directly (so `verusfmt.args(walk_vstd)` or similar).

Reply:
Makes sense.